File: Constructor Open Cup 2024/c.c
#include <stdio.h>

int max_a1(int n, int *arr)
{
    int max_val = arr[0];
    int odd_sum = 0, even_sum = 0;

    // Calculate the sum of elements at odd and even indices
    for (int i = 1; i < n; i++)
    {
        if (i % 2 == 0)
            even_sum += arr[i];
        else
            odd_sum += arr[i];
    }

    // Choose the maximum between the initial value and the sum of odd indices
    if (odd_sum > max_val)
        max_val = odd_sum;

    // Return the maximum value of a1
    return max_val;
}

int main()
{
    int t;
    scanf("%d", &t);

    while (t--)
    {
        int n;
        scanf("%d", &n);
        int arr[n];

        // Input array elements
        for (int i = 0; i < n; i++)
        {
            scanf("%d", &arr[i]);
        }

        // Calculate and print the maximum possible value of a1
        printf("%d\n", max_a1(n, arr));
    }

    return 0;
}

File: Constructor Open Cup 2024/c.py
test_cases = int(input())

for _ in range(test_cases):
  n = int(input())
  a = list(map(int, input().split()))

  even = sorted(a)
  odd = sorted(a, reverse=True)

  for i in range(n // 2):
    if i % 2 == 0:
      a[i*2] = even[i]
      a[i*2+1] = even[n-i-1]
    else:
      a[i*2] = odd[i]  
      a[i*2+1] = odd[n-i-1]

  print(a[0])
